packages/webcert/src/feature/certificate/Inputs/UeUncertainDate.tsx

```diff
@@ -0,0 +1,127 @@
+import { CertificateDataElement, Dropdown, ConfigureUeUncertainDate, QuestionValidationTexts, ValueDate } from '@frontend/common'
+import { ConfigUeDropdownItem } from '@frontend/common/src/types/certificate'
+import React from 'react'
+import styled from 'styled-components/macro'
+import { updateCertificateDataElement } from '../../../store/certificate/certificateActions'
+import { useAppDispatch } from '../../../store/store'
+
+const ValidationWrapper = styled.div`
+  flex: 0 !important;
+  flex-basis: 100% !important;
+  padding-bottom: 16px;
+  margin-top: 0;
+`
+
+export interface Props {
+  hasValidationError?: boolean
+  disabled?: boolean
+  question: CertificateDataElement
+}
+
+const UeUncertainDate: React.FC<Props> = (props) => {
+  const { question, hasValidationError, disabled } = props
+  const config = question.config as ConfigureUeUncertainDate
+  const dispatch = useAppDispatch()
+  const [selectedYear, setSelectedYear] = React.useState(getYear((question.value as ValueDate).date))
+  const [selectedMonth, setSelectedMonth] = React.useState(getMonth((question.value as ValueDate).date))
+  const [disabledMonth, setDisabledMonth] = React.useState(
+    disabled || getYear((question.value as ValueDate).date) === '0000' || getYear((question.value as ValueDate).date) === ''
+  )
+
+  const years: ConfigUeDropdownItem[] = [{ id: '', label: 'Ange år' }]
+
+  if (config.unknownYear) years.push({ id: '0000', label: '0000 (ej känt)' })
+  config.allowedYears.forEach((item) => {
+    years.push({ id: item, label: item })
+  })
+
+  const months: ConfigUeDropdownItem[] = [{ id: '', label: 'Ange månad' }]
+  if (config.unknownMonth) months.push({ id: '00', label: '00 (ej känt)' })
+  for (let i = 1; i <= 12; i++) {
+    const strMonth = '0' + i.toString()
+    const month = strMonth.substring(strMonth.length - 2)
+    months.push({ id: month, label: month })
+  }
+
+  const handleYearChange = (value: string) => {
+    setSelectedYear(value)
+    if (value === '') setSelectedMonth('')
+    if (value === '0000') setSelectedMonth('00')
+    setDisabledMonth(disabled || value === '0000' || value === '')
```

Comment:
Probably just me being confused, but which one is month set to? Disabled, 0000 or ''?

Reply:
month is set to not selected ('', ange månad) if year is not selected ('', ange år)
month is set to unknown ('00') if year is set to unknown ('0000')
the month control is disabled if either the whole control is disabled, or if year is not selected ('') or if year is unknown ('0000')